### PR TITLE
CDSK-953 - fix Interpolate has no length exception

### PR DESCRIPTION
**Pull request by warcholprzemo**
Interpolate is not Iterable so we can resolve Interpolate to string and continue do a stuff in ShellCommand._describe

### CODE REVIEW

File: master/buildbot/steps/shell.py [hunk 1]
```diff
@@ -200,6 +201,8 @@ def _describe(self, done=False):
             words = self.command
             if isinstance(words, (str, unicode)):
                 words = words.split()
+            elif isinstance(words, Interpolate):
+                return str(words)
```
Comment (campos-ddc) [inline master/buildbot/steps/shell.py]: Are you sure it is safe to return a string here?

The rest of code is always returning lists of strings.

Comment (warcholprzemo) [reply master/buildbot/steps/shell.py]: You're right. I did not notice a type for return value.

File: master/buildbot/steps/shell.py [hunk 2]
```diff
@@ -200,6 +201,8 @@ def _describe(self, done=False):
             words = self.command
             if isinstance(words, (str, unicode)):
                 words = words.split()
+            elif isinstance(words, Interpolate):
+                return [str(words)]
```
Comment (Mowinski) [inline master/buildbot/steps/shell.py]: I do not sure this solution. Now we can not display just a part of interpolation and in a description, there will be always full version and as I can see, sometimes there is a very long shell command. I think we should consider how to split `Interpolate` command.

@campos-ddc  is it ok right now or we should change it?

File: master/buildbot/test/unit/test_steps_shell.py [hunk 3]
```diff
@@ -71,6 +72,11 @@ def test_constructor_args_validity(self):
                     lambda: shell.ShellCommand('build', "echo Hello World",
                             wrongArg1=1, wrongArg2='two'))
 
+    def test_describe_with_command_Interpolate(self):
```
Comment (Mowinski) [inline master/buildbot/test/unit/test_steps_shell.py:76]: Right now, you are also testing how casting work in Interpolate class. I think you should separate command to a string variable, and pass it as an argument to `Interpolate` class and also assert that `self.describe()` return this string in array form.

**Review by Mowinski — CHANGES_REQUESTED** [review]

File: master/buildbot/test/unit/test_steps_shell.py (hunk 3 above)
Comment (warcholprzemo) [reply master/buildbot/test/unit/test_steps_shell.py:76]: Done. We check `expected_output` with an output from `step.describe()`

File: master/buildbot/steps/shell.py (hunk 2 above)
Comment (warcholprzemo) [reply master/buildbot/steps/shell.py]: After talking with Kamil I've changed `str(Interpolate)` to resolve Interpolate with `.getRenderingFor(self.build).result`

**Review by campos-ddc — APPROVED** [review]

**Pull request merged**